### ISSUE TITLE
`$app->environment` with parameter

### ISSUE THREAD

**Issue by brendt** [nunomaduro/larastan#91]
`Illuminate\Contracts\Foundation\Application` defines the `environment` function as follows

```php
/**
 * Get or check the current application environment.
 *
 * @return string
 */
public function environment();
```

The concrete `Application` implementation in `Illuminate\Foundation\Application` implements it like so:

```php
/**
 * Get or check the current application environment.
 *
 * @return string|bool
 */
public function environment()
{
    if (func_num_args() > 0) {
        $patterns = is_array(func_get_arg(0)) ? func_get_arg(0) : func_get_args();

        return Str::is($patterns, $this['env']);
    }

    return $this['env'];
}
```

Meaning you're able to pass arguments to the function, though it's never documented 🙄

Not sure whether a docblock or optional paramter must be added in the core, or if you want to fix it in this package. Anyways, when doing this:

```php
$this->app->environment('local')
```

There's this error:

```php
Method Illuminate\Contracts\Foundation\Application::environment() invoked with 1 parameter, 0 required.
```

**Comment by szepeviktor** [comment]
@brendt Submit a Laravel PR!

**Comment by brendt** [comment]
A PR addressing this issue has already been rejected: https://github.com/laravel/framework/pull/23533

**Comment by szepeviktor** [comment]
@nunomaduro Could it be that it'll be the next `ignoreErrors:`?

**Comment by nunomaduro** [comment]
@szepeviktor @brendt I will write an extension for this.

**Comment by claystation** [comment]
Any news? I commented on #184 but now i see there is also this issue.

**Comment by spawnia** [comment]
This is resolved as of Laravel `5.8`, see https://github.com/laravel/framework/commit/f0b7f49a7f6d15591859e7ecb06c090b02c68df6#diff-7b18a52eceff5eb716c1de268e98d55d
